File: front/src/services/address/AddressService.ts
import type {
  ConnectionAddressConsent,
  CreateAddressInput,
  UpdateAddressInput,
  UserAddress,
} from './types.ts'

export interface AddressService {
  myAddresses(): Promise<UserAddress[]>
  addressConsentsForConnection(connectionId: string): Promise<ConnectionAddressConsent[]>
  createAddress(input: CreateAddressInput): Promise<UserAddress>
  updateAddress(id: string, input: UpdateAddressInput): Promise<UserAddress>
  deleteAddress(id: string): Promise<boolean>
  grantAddressConsent(connectionId: string, addressId: string): Promise<ConnectionAddressConsent>
  revokeAddressConsent(connectionId: string): Promise<ConnectionAddressConsent>
}

File: back/src/main/kotlin/com/jvmvstv_v/back/address/service/AddressService.kt
package com.jvmvstv_v.back.address.service

import com.jvmvstv_v.back.address.model.ConnectionAddressConsent
import com.jvmvstv_v.back.address.model.CreateAddressInput
import com.jvmvstv_v.back.address.model.UpdateAddressInput
import com.jvmvstv_v.back.address.model.UserAddress
import java.util.UUID

interface AddressService {
    fun myAddresses(): List<UserAddress>
    fun consentsForConnection(connectionId: UUID): List<ConnectionAddressConsent>
    fun createAddress(input: CreateAddressInput): UserAddress
    fun updateAddress(id: UUID, input: UpdateAddressInput): UserAddress
    fun deleteAddress(id: UUID): Boolean
    fun grantConsent(connectionId: UUID, addressId: UUID): ConnectionAddressConsent
    fun revokeConsent(connectionId: UUID): ConnectionAddressConsent
}
